src/paymentservice/charge.js

```diff
@@ -1,24 +1,39 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
-const {context, propagation, trace, metrics} = require('@opentelemetry/api');
+const { context, propagation, trace, metrics } = require('@opentelemetry/api');
 const cardValidator = require('simple-card-validator');
 const { v4: uuidv4 } = require('uuid');
 
 const { OpenFeature } = require('@openfeature/server-sdk');
-const { FlagdProvider} = require('@openfeature/flagd-provider');
+const { FlagdProvider } = require('@openfeature/flagd-provider');
 const flagProvider = new FlagdProvider();
 
 const logger = require('./logger');
 const tracer = trace.getTracer('paymentservice');
 const meter = metrics.getMeter('paymentservice');
-const transactionsCounter = meter.createCounter('app.payment.transactions')
+const transactionsCounter = meter.createCounter('app.payment.transactions');
+
+const TENANT_LEVEL = ['gold', 'silver', 'bronze'];
+
+/** Return random element from given array */
+function random(arr) {
+  const index = Math.floor(Math.random() * arr.length);
+  return arr[index];
+}
 
 module.exports.charge = async request => {
   const span = tracer.startSpan('charge');
 
   await OpenFeature.setProviderAndWait(flagProvider);
+
   if (await OpenFeature.getClient().getBooleanValue("paymentServiceFailure", false)) {
-    throw new Error("PaymentService Fail Feature Flag Enabled");
+    // 75% chance to fail with version 350.10 span tag
+    if (Math.random() < 0.75) {
+      span.setAttributes({ 'app.version': '350.10', 'app.tenant.level': random(TENANT_LEVEL) });
```

Comment:
To set `service.version` we are following the OTel semantic conventions.
What would `app.version` be? And is there any case where this value would be different from `service.version`?

Currently this is being set here: https://github.com/open-telemetry/opentelemetry-demo/blob/main/.env#L40

Reply:
`app.version` is specific to the `paymentservice` whereas (I believe) `service.version` applies to all services running the application. `app.version` has two versions where one will error and the other will not. Here is a visual that shows this off better (I hope) :-)

<img width="456" alt="image" src="https://github.com/user-attachments/assets/6a7f0a11-d85e-4094-aff4-701ee71e4942">

Reply:
I believe both are referring to the same `version`.
In the demo it happens to be the same, as we opted to set it globally, but each service could have its own version.

To set the app version following the OTel semantic conventions, we need to use `service.version`.

Reply:
Moving to use `app.variant` instead with the using values `A` & `B`. Removes confusion around versions. Service will still maintain the version set in the resource attributes.

Reply:
I believe that in this case is not matter of key, but actually meaning.
You will call it `app.variant` but it will still be a version identifier.

Let me bring some folks to the discussion, I'd like to hear more opinions.
@rogercoll, @puckpuck and @mviitane WDYT?

Reply:
I don't think `app.variant` makes sense here. Are we trying to demonstrate a fictional state that a failure only happens on a certain app variant or deployed version? This demonstration mechanism should be handled within the vendor's demo or some other means ([DemoMonkey](https://www.demomonkey.net/) comes to mind). I wouldn't want to give a user the impression that we deployed 2 different variants/versions of the app.

This should be called `app.<featureflagname>` and have its value be what the feature flag returned.

Reply:
`app.variant` removed from PR.